src/util/AbstractHelper.sol

```diff
@@ -0,0 +1,337 @@
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function sellDbrRepayAndWithdrawNativeEthOnBehalf(
+        IMarket market, 
+        uint dolaAmount, 
+        uint minDolaFromDbr,
+        uint dbrAmountToSell, 
+        uint collateralAmount, 
+        uint deadline, 
+        uint8 v, 
+        bytes32 r, 
+        bytes32 s) 
+        external 
+    {
+        //Repay
+        sellDbrAndRepayOnBehalf(market, dolaAmount, minDolaFromDbr, dbrAmountToSell);
+
+        //Withdraw
+        withdrawNativeEthOnBehalf(market, collateralAmount, deadline, v, r, s);
+    }
+
+    /**
+    @notice Helper function for depositing native eth to WETH markets
+    @param market The WETH market to deposit to
+    */
+    function depositNativeEthOnBehalf(IMarket market) public payable {
+        require(address(market.collateral()) == address(WETH), "Not an ETH market");
+        WETH.deposit{value:msg.value}();
+        WETH.approve(address(market), msg.value);
+        market.deposit(msg.sender, msg.value);
+    }
+
+    /**
+    @notice Helper function for depositing native eth to WETH markets before borrowing on behalf of the depositor
+    @param market The WETH market to deposit to
+    @param borrowAmount The amount to borrow on behalf of the depositor
+    @param deadline Deadline of the signature
+    @param v V parameter of the signature
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function depositNativeEthAndBorrowOnBehalf(IMarket market, uint borrowAmount, uint deadline, uint8 v, bytes32 r, bytes32 s) public payable {
+        require(address(market.collateral()) == address(WETH), "Not an ETH market");
+
+        //Deposit native eth
+        WETH.deposit{value:msg.value}();
+        WETH.approve(address(market), msg.value);
+        market.deposit(msg.sender, msg.value);
+
+        //Borrow Dola
+        market.borrowOnBehalf(msg.sender, borrowAmount, deadline, v, r, s);
+        DOLA.transfer(msg.sender, borrowAmount);
+    }
+
+    /**
+    @notice Helper function for withdrawing to native eth
+    @param market WETH market to withdraw collateral from
+    @param collateralAmount Amount of collateral to withdraw
+    @param deadline Deadline of the signature
+    @param v V parameter of the signature
+    @param r R parameter of the signature
+    @param s S parameter of the signature
+    */
+    function withdrawNativeEthOnBehalf(
+        IMarket market,
+        uint collateralAmount,
+        uint deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s)
+        public
+    {
+        market.withdrawOnBehalf(msg.sender, collateralAmount, deadline, v, r, s);
+
+        IERC20 collateral = IERC20(market.collateral());
+        require(address(collateral) == address(WETH), "Not an ETH market");
+        WETH.withdraw(collateralAmount);
+
+        payable(msg.sender).transfer(collateralAmount);      
```

Comment:
Due to EIP-2929 this may run out of gas if `msg.sender` is a contract with logic in fallback (e.g. Gnosis Safe)
Would suggest using `.call()` instead.

See:
https://help.safe.global/en/articles/5249851-why-can-t-i-transfer-eth-from-a-contract-into-a-safe

Reply:
Fixed in [c086d8c](https://github.com/InverseFinance/FiRM/pull/33/commits/c086d8c89c1c094b1ac408940fcadfd3d363e7cb)